mlir/lib/Bindings/Python/IRCore.cpp

```diff
@@ -3718,6 +3724,36 @@ void mlir::python::populateIRCore(py::module &m) {
             mlirValueReplaceAllUsesOfWith(self.get(), with.get());
           },
           kValueReplaceAllUsesWithDocstring)
+      .def(
+          "replace_all_uses_except",
+          [](PyValue &self, PyValue &with, PyOperation &exception) {
```

Comment:
sorry i didn't notice - you can just do here (and below)
```suggestion
          [](MlirValue self, MlirValue with, MlirOperation exception) {
```

all of this magic over here in [PybindAdaptors.h](https://github.com/llvm/llvm-project/blob/825d9f6e0cea21c50a767e80be5ba80479d65f0e/mlir/include/mlir/Bindings/Python/PybindAdaptors.h#L66-L202) will take care of it for you.

Reply:
I note that some existing code such as `replace_all_uses_with` also uses `PyValue &self, PyValue &with`, should I look at making that consistent in a future PR?

Reply:
I took a quick look in `IRCore.cpp` before making the suggestion it seems those uses are because the various fields/methods of those classes are in fact exercised. Here, where you're just immediately calling the C API, you don't need any of that functionality. I mean ultimately this is a nit - it's up to you - you can save a couple of lines but it makes no difference (the magic in `PybindAdaptors.h` does exactly what you're doing here - `.get()`).